docs/architecture/adr-071-proposer-based-timestamps.md

```diff
@@ -0,0 +1,458 @@
+With these timestamps removed, the commit time will instead be stored in the `Commit` struct.
+
+`Commit` currently contains the following fields: 
+
+```go
+type Commit struct {
+	Height     int64       `json:"height"`
+	Round      int32       `json:"round"`
+	BlockID    BlockID     `json:"block_id"`
+	Signatures []CommitSig `json:"signatures"`
+}
+```
+
+`Commit` will be updated to contain the following fields: 
+
+```go
+type Commit struct {
+	Height     int64       `json:"height"`
+	Round      int32       `json:"round"`
+	Timestamp   time.Time   `json:"round"`
+	BlockID    BlockID     `json:"block_id"`
+	Signatures []CommitSig `json:"signatures"`
+}
+```
+
+### Changes to `Vote` messages
+
+`Precommit` and `Prevote` messages use a common [Vote struct](https://github.com/tendermint/tendermint/blob/a419f4df76fe4aed668a6c74696deabb9fe73211/types/vote.go#L50).
+This struct currently contains a timestamp.
+This timestamp is set using the [voteTime](https://github.com/tendermint/tendermint/blob/e8013281281985e3ada7819f42502b09623d24a0/internal/consensus/state.go#L2241) function and therefore vote times correspond to the validator’s local time.
+For precommits, this timestamp is used to construct the [CommitSig that is included in the block in the LastCommit](https://github.com/tendermint/tendermint/blob/e8013281281985e3ada7819f42502b09623d24a0/types/block.go#L754) field.
+For prevotes, this field is unused.
+Proposer-based timestamps will use the [RoundState.Proposal](https://github.com/tendermint/tendermint/blob/c3ae6f5b58e07b29c62bfdc5715b6bf8ae5ee951/internal/consensus/types/round_state.go#L76) timestamp to construct the `signedBytes` `CommitSig`.
+This timestamp is therefore no longer useful and will be dropped.
+
+`Vote` currently contains the following fields:
+
+```go
+type Vote struct {
+	Type             tmproto.SignedMsgType `json:"type"`
+	Height           int64                 `json:"height"`
+	Round            int32                 `json:"round"`    
+	BlockID          BlockID               `json:"block_id"` // zero if vote is nil.
+	Timestamp        time.Time             `json:"timestamp"`
+	ValidatorAddress Address               `json:"validator_address"`
+	ValidatorIndex   int32                 `json:"validator_index"`
+	Signature        []byte                `json:"signature"`
+}
+```
+
+`Vote` will be changed to contain the following fields:
+
+```go
+type Vote struct {
+	Type             tmproto.SignedMsgType `json:"type"`
+	Height           int64                 `json:"height"`
+	Round            int32                 `json:"round"` // proposal/vote round   
+	ForBlock         bool                  `json:"for_block"` // false if vote is nil.
+	ValidatorIndex   int32                 `json:"validator_index"`
+	Signature        []byte                `json:"signature"`
+}
+```
+
+`BlockID` is also dropped as it is also stored by all validators.
```

Comment:
should we keep the scope of this ADR to proposer timestamps? We can do clean up at a later stage or at the same time. the scope of removing some of these things can be large.

Reply:
I am not sure if we can remove `BlockID` from the vote. How a process is then able to say for what block it is voting?

Reply:
This may be overkill for this change, I'm not sure. Validators receive a proposal each round and store the proposal. The proposal contains a `BlockID`. The signed bytes used to generate the vote signature contains a hash of the `BlockID`. Since all validators are already saving the `BlockID` in the propose message, they should be able to use their locally stored `BlockID` information to generate the signedBytes and validate the vote signature against that instead of also requiring it be part of the votes message.

Reply:
Removed these changes. We have an additional issue in progress related to removing additional fields from the block, #4835. can discuss these changes along with that work.